lumicks/pylake/kymo.py

```diff
@@ -729,3 +729,99 @@ def line_timestamp_ranges_factory(_, exclude=bool):
+    name : str
+        Kymo name.
+    reduce : callable
+        The function which is going to reduce multiple pixels into one. The default is
+        :func:`numpy.mean`, but :func:`numpy.max` could also be appropriate for some cases.
+    one_color_default : { "red", "green", "blue" }
+        A `Kymo` has three color channels "red", "green", and "blue". A `CorrelatedStack` might
+        have three or only one color channel. The `one_color_default` sets the channel to be used
+        in a `Kymo` created from a single-color `CorrelatedStack`. The `one_color_default` is only
+        evaluated for single-color `CorrelatedStack`s.
+    """
+    # Ensure constant frame rate of the whole stack
+    ts_ranges = np.array(corrstack.frame_timestamp_ranges)
+    line_times = np.diff(ts_ranges[:, 0])
+    line_time = line_times[0]
+    if not np.all(line_times == line_time):
+        raise ValueError("The frame rate of the images of the correlated stack is not constant.")
+    line_time_s = line_time * 1e-9
+
+    # Ensure constant exposure time of the whole stack
+    exp_times = ts_ranges[:, 1] - ts_ranges[:, 0]
+    exp_time = exp_times[0]
+    if not np.all(exp_times == exp_time):
+        raise ValueError("The exposure time of the images of the correlated stack is not constant.")
+    exp_time_s = exp_time * 1e-9
+
+    # Start timestamp of the kymo is start timestamp of first image
+    start = ts_ranges[0, 0]
+
+    # Ensure correlated stack has proper tether
+    if not corrstack.src._tether:
+        raise ValueError("The correlated stack does not have a tether.")
+    (x1, y1), (x2, y2) = corrstack.src._tether.ends
+    if np.floor(y1) != np.floor(y2):
+        raise ValueError("The correlated stack is not aligned along the tether axis.")
+
+    # Extract the kymograph data along the line of the tether
+    # Determine next rounded odd width minus 1 -> halfwidth = 0, 1, 2, ...
+    halfwidth = int(width // 2)
+    if halfwidth < 0:
+        raise ValueError("The requested `halfwidth` must not be negative.")
+    xmin = int(np.floor(x1))
+    xmax = int(np.floor(x2)) + 1
+    ymin = int(np.floor(y1)) - halfwidth
+    ymax = int(np.floor(y2)) + halfwidth + 1
+    if ymin < 0 or ymax > corrstack.shape[1]:
+        raise ValueError(
+            "The requested `width` of the line exceeds the size of the correlated stack images."
+        )
+    kymostack = corrstack.crop_by_pixels(xmin, xmax, ymin, ymax)
+    image = kymostack.get_image(channel="rgb")  # time, (y,) x, (c)
+    if halfwidth > 0:
+        image = reduce(image, axis=1)  # time, x, (c)
+    image = np.swapaxes(image, 0, 1)  # x, time, (c)
+
+    color_format = "rgb"
+    if image.ndim == 2:
+        if one_color_default not in ["red", "green", "blue"]:
+            raise ValueError(
+                'Invalid `one_color_default` format. Allowed values are "red", "green" or "blue".'
+            )
+        color_format = one_color_default[0]
```

Comment:
Just an option (I usually find it easier to reason about something when there's only a single assignment):
```suggestion
    try:
        color_options = {"red": "r", "green": "g", "blue": "b"}
        color_format = color_options[one_color_default] if image.ndim == 2 else "rgb"
    except KeyError:
        raise ValueError(
            'Invalid `one_color_default` format. Allowed values are "red", "green" or "blue".'
        )
```

I am wondering whether this `one_color_default` parameter may be confusing though. I'm doubting whether it should just always be "rgb" instead and the data tiled out to be three channels if it's just one channel. What do you think?

Reply:
I also was not happy with the parameter `one_color_default` and I like the idea of "triplicating" the data of a correlated stack with only one channel and default to "rgb". I will go ahead with implementing this. (I would prefer your code suggestion if I would keep `one_color_default`!)